### ISSUE TITLE
Probability Distributions in Keras-Core

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras#18435]
Hello everyone!  I recently wrote https://github.com/keras-team/keras-io/pull/1440 and I'd like to make this example fully keras-core friendly.  This requires implementations of probability distributions at some sort.

While reviewing TensorFlow probability distributions & torch.distribution I noticed that the distribution classes house many methods, not just a `sample()` method!

Roughly they share (from TFP):
- log_prob()
- log_survival_function()
- cdf()
- prob(value)
- log_cdf()
- covariance()
- cross_entropy()
- entropy()
- mean()
- mode()
- stddev()
- unnormalized_log_prob()
- variance()
- quantile()
- ability to operate via logits or probs
- differentiable()
- some other shared interface methods

Reference:
https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Distribution

---

I personally think we should introduce some probability distribution class to Keras core!  We can back them with numpy()/tfp.Categorical()/torch.distributions().

It might be a bit of work to make the APIs match - but seems worthwhile.  Once we agree on a class design, I'd love to take a stab at adding one!
